### PR TITLE
made changes to the interfaces and lambdas slide.

### PR DESCRIPTION
**Pull request by miscbits**
@nhu313 and I went through the slides to review before Monday's class and made some changes we thought would improve them. I also added some cute animals to split up the slides into two parts.

### CODE REVIEW

File: md/InterfacesandLambdas.md [hunk 1]
```diff
@@ -66,17 +70,20 @@ public int compareTo(String anotherString)
...
-      System.out.println("Name=" + s.getName() + ", order=" + s.getPowerLevel());
+      System.out.println("Name=" + s.getName() 
+          + ", order=" + s.getPowerLevel());
 
+    ...
   }
 ```
+-
+
+    * There is no way for this method to know how to sort these custom objects by default unless this object implements Comparable.
```
Comment (Git-Leon) [inline md/InterfacesandLambdas.md]: There is no way for this method to know how to sort these custom objects by default unless this ~object~ **class** implements Comparable.

File: md/InterfacesandLambdas.md [hunk 2]
```diff
@@ -1,32 +1,31 @@
...
-package io.zipcoder;
+public class RandomQuote implements SmartQuotes { ... }
 
-public interface SmartQuotes {
-     String selectRandomQuote();
-}
 ```
 -
 
+* Within this class you _must_ implement all methods within the interface
```
Comment (Git-Leon) [inline md/InterfacesandLambdas.md]: Within this class you _must_ implement all methods ~within~ **declared by** the interface

File: md/InterfacesandLambdas.md [hunk 3]
```diff
@@ -280,13 +295,14 @@ public interface CaptainAmericasShield {
 public class Vision implements Mjolnir, CaptainAmericasShield {
 
   /**
-   * Here we have a class that implements two interfaces with default implementations
-   * the compiler has no way of implicitly who should be called
-   * so we have to explicitly state it.
+   * Here we have a class that implements two interfaces with
```
Comment (Git-Leon) [inline md/InterfacesandLambdas.md]: this is a strange grouping of 3 bullets...

Comment (Git-Leon) [inline md/InterfacesandLambdas.md]: * Here we have a class that implements two interfaces with default implementations.
* The compiler has no way of implicitly **knowing** who should be called so we have to explicitly state it.

File: md/InterfacesandLambdas.md (hunk 2 above)
Comment (miscbits) [reply md/InterfacesandLambdas.md]: 👍

File: md/InterfacesandLambdas.md (hunk 1 above)
Comment (miscbits) [reply md/InterfacesandLambdas.md]: 👍

**Pull request merged**